transpiler/src/ast/core/intermediate-ast/builders/UseCase/UseCaseNodeBuilder.ts

```diff
@@ -0,0 +1,49 @@
+import { IntermediateASTTree } from '../../IntermediateASTTree.js';
+import { TNodeMetadata } from '../../nodes/IntermediateASTNode.js';
+import { ParameterListNode } from '../../nodes/ParameterList/ParameterListNode.js';
+import { UseCaseExecuteNode } from '../../nodes/UseCase/UseCaseExecuteNode.js';
+import { UseCaseIdentifierNode } from '../../nodes/UseCase/UseCaseIdentifierNode.js';
+import { UseCaseNode } from '../../nodes/UseCase/UseCaseNode.js';
+import { IBuilder } from '../IBuilder.js';
+
+export class UseCaseNodeBuilder implements IBuilder<UseCaseNode> {
+  private useCaseNode: UseCaseNode;
+  private identifierNode: UseCaseIdentifierNode;
+  private executeNode: UseCaseExecuteNode;
+  private parameterListNode: ParameterListNode;
+  private intermediateASTTree: IntermediateASTTree;
+
+  constructor(intermediateASTTree: IntermediateASTTree, metadata?: TNodeMetadata) {
+    this.intermediateASTTree = intermediateASTTree;
+    this.useCaseNode = new UseCaseNode(metadata);
+  }
+
+  public withIdentifier(useCaseIdentifierNode: UseCaseIdentifierNode): UseCaseNodeBuilder {
+    this.identifierNode = useCaseIdentifierNode;
+    const useCaseName = useCaseIdentifierNode.getIdentifierName();
```

Comment:
I've also done this, but perhaps we should add getClassName func in ClassTypeNode which returns the className after searching for the identifier? so we don't have to do it on each builder